source/common/http/path_utility.h

```diff
@@ -24,5 +29,21 @@ class PathUtil {
   static absl::string_view removeQueryAndFragment(const absl::string_view path);
 };
 
+using Transformation = std::function<absl::optional<std::string>(absl::string_view)>;
+
+class PathTransformer {
+public:
+  PathTransformer(envoy::type::http::v3::PathTransformation operations);
+
+  absl::optional<std::string> transform(const absl::string_view original_path);
```

Comment:
could you add some tests for `transform` as well (ingest a PathTransformer config) please?
(maybe re-use some of the previous tests to also run through a config and compare)

Reply:
Thanks, test added.